src/core/types/Service.ts

```diff
@@ -161,6 +165,28 @@ export interface GetTokenAllowanceProps {
   spenderAddress: string;
 }
 
+export interface ApproveDepositProps {
+  network: Network;
+  accountAddress: Address;
+  tokenAddress: Address;
+  amount: Wei;
+  vault: Vault;
+}
+
+export interface ApproveZapOutProps {
+  network: Network;
+  accountAddress: Address;
+  amount: Wei;
+  vault: Vault;
+}
+
+export interface GetVaultAllowanceProps {
+  network: Network;
+  accountAddress: string;
+  tokenAddress: string;
+  vault: { address: string; token: string };
```

Comment:
We should try to use a flat object when possible and not nested objects.

Reply:
sure will change that

Reply:
solved on fff3e86f6179d4e466b091c8f555551d66f809ef